packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -56,6 +59,34 @@ class NetworkController extends DisposableController
     );
     subscribeToFilterChanges();
   }
+  List<DartIOHttpRequestData>? _httpRequests;
+
+  String? exportAsHarFile() {
+    final exportController = ExportController();
+
+    _httpRequests =
+        filteredData.value.whereType<DartIOHttpRequestData>().toList();
+
+    if (_httpRequests!.isEmpty) {
+      debugPrint('No valid request data to export');
+      return '';
+    }
+
+    try {
+      if (_httpRequests != null && _httpRequests!.isNotEmpty) {
+        // Build the HAR object
+        final har = buildHar(_httpRequests!);
+        debugPrint('data is ${json.encode(har)}');
+        return exportController.downloadFile(
+          json.encode(har),
+          type: ExportFileType.har,
+        );
+      }
+    } catch (ex) {
```

Comment:
nit: rename exception to `e`. Out of curiosity, what sorts of exceptions were you seeing? I noticed in other parts of the DevTools codebase that we are not needing to wrap these `downloadFile` method calls in a try catch

Reply:
done, the exceptions were during the building of har when i was developing, I kept this for safety